metrics-exporter-prometheus/src/exporter/http_listener.rs

```diff
@@ -29,31 +58,52 @@ impl HttpListeningExporter {
                     continue;
                 }
             };
-
-            let is_allowed = self.allowed_addresses.as_ref().map_or(true, |addrs| {
-                stream.peer_addr().map_or_else(
-                    |e| {
-                        warn!(error = ?e, "Error obtaining remote address.");
-                        false
-                    },
-                    |peer_addr| {
-                        let remote_ip = peer_addr.ip();
-                        addrs.iter().any(|addr| addr.contains(&remote_ip))
-                    },
-                )
-            });
-
-            self.process_stream(stream, is_allowed).await;
+            self.process_tcp_stream(stream).await;
         }
     }
 
-    async fn process_stream(&self, stream: TcpStream, is_allowed: bool) {
+    async fn process_tcp_stream(&self, stream: TcpStream) {
+        let is_allowed = self.check_tcp_allowed(&stream);
         let handle = self.handle.clone();
         let service = service_fn(move |req: Request<body::Incoming>| {
             let handle = handle.clone();
             async move { Ok::<_, hyper::Error>(Self::handle_http_request(is_allowed, &handle, &req)) }
         });
 
+        tokio::spawn(async move {
+            if let Err(err) =
+                HyperHttpBuilder::new().serve_connection(TokioIo::new(stream), service).await
+            {
+                warn!(error = ?err, "Error serving connection.");
+            }
+        });
+    }
+
+    fn check_tcp_allowed(&self, stream: &TcpStream) -> bool {
+        if let Some(addrs) = &self.allowed_addresses {
+            if let Ok(peer_addr) = stream.peer_addr() {
+                return addrs.iter().any(|addr| addr.contains(&peer_addr.ip()));
+            }
+        }
+        true
+    }
+
+    #[cfg(feature = "uds-listener")]
+    async fn serve_uds(&self, listener: &UnixListener) -> Result<(), std::io::Error> {
+        loop {
+            let (stream, _) = listener.accept().await?;
```

Comment:
I think we'd want this to match how we do it in the TCP listener, where we log a warning but still keep accepting connections.